packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -339,25 +339,30 @@ class AppDomain extends Domain {
       );
     }
 
+    bool supportsRestart = hotMode ? device.supportsHotMode : device.supportsRestart;
```

Comment:
So some other notes on this. Hot mode needs to keep a state machine (I am in the process of implementing this now) of what is currently supported by the running program:

- restart is always supported

after a start/restart:
- initially hot reloaded is supported
- if the isolate hits an unhandled exception, hot reload is disabled until the application is restarted

The IDE will need to react by dimming/undimming the reload button when it can't be done. How do you want the runner to advertise to the IDE the changes in these states over time?